app/main.py

```diff
@@ -1,62 +1,132 @@
-async def update_todo(updated_todo: Todo):
-    for todo in global_todos:
-        if todo.id == updated_todo.id:
-            todo.id = updated_todo.id
-            todo.name = updated_todo.name
-
-            return {f"'message': Updated todo {updated_todo.name}'"}
+from fastapi import FastAPI, Query, Path, BackgroundTasks
+# from fastapi.testclient import TestClient
+from pydantic import BaseModel
+import numpy as np
+import requests
+import appliance
+import InstallAPIcalls
+import time
+from fastapi.middleware.wsgi import WSGIMiddleware
+from InstallGraphing import app as ElectricityGraph
 
-    return {f"'message': 'Todo not found'"}
+app = FastAPI()
 
 
-# Delete Todo by id
-@app.delete("/todos/{todo_id}")
-async def delete_todo(todo_id: int):
-    global global_todos
-    updated_todos = filter(lambda todo: todo.id != todo_id, global_todos)
-    global_todos[:] = list(updated_todos)
 
-    return {"message": "Todo successfully deleted"}
+users = []
+customers = []
+elec_dicts = []
+
+
+# @app.get("/")
+# def read_root():
+#     return {"Hello": "World"}
+
+# client = TestClient(app)
+
+@app.post("/users/")
+async def read_user_bayou(
+        utility: str = "speculoos_power",
+        email: str = "QuickStart@bayou.energy"
+):
+    bayou_user = {
+        "utility": utility,
+        "email": email
+    }
+
+    users.append(bayou_user)
+    return users
+
+
+@app.get("/users/{user_id}")
+async def read_user(user_id: int):
+    return users[user_id]
+
+
+# This is to create a new customer in Bayou
+@app.post("/new_customers/")
+def bayou_customer_generation(
+        user_id: int
+):
+    bayou_user = users[user_id]
+    utility = bayou_user["utility"]
+    email = bayou_user["email"]
+
+    check = InstallAPIcalls.UtilityCheck(utility)
+    if check:
+        return check
+    else:
+        customer = InstallAPIcalls.BayouAPICustomer(utility, email)
+    if not customer["has_filled_credentials"]:
+        customer_link = InstallAPIcalls.BayouAPICustomerLink(customer)
+        return customer_link
+    else:
+        customers.append(customer)
+        return customer
+
+
+# This is to pull customer data from Bayou
+@app.post("/customers/")
+def bayou_customer_pull(user_id: int):
+    bayou_user = users[user_id]
+    utility = bayou_user["utility"]
+    email = bayou_user["email"]
+
+    check = InstallAPIcalls.UtilityCheck(utility)
+    if check:
+        return check
+    else:
+        customer = InstallAPIcalls.BayouAPICustomer2(utility, email)
+    if not customer["has_filled_credentials"]:
+        customer_link = InstallAPIcalls.BayouAPICustomerLink(customer)
+        return customer_link
+    else:
+        customers.append(customer)
+        return customer
+
+
+@app.put("/customers/{customer_id}")
+def bayou_customer_credentials(customer_id: int):
+    customer = customers[customer_id]
+    customer_link = InstallAPIcalls.BayouAPICustomerLink(customer)
+    return customer_link
+
+
+@app.post("/electricity_data/")
```

Comment:
Maybe we can rename this endpoint to just `electricity` unless we might need that for something else?

Reply:
Got it!